tracy-edit/src/OfflineSymbolResolverDbgHelper.cpp

```diff
@@ -0,0 +1,142 @@
+                                 SymbolEntryList& resolvedEntries)
+    {
+        ULONG64 moduleBase = SymLoadModuleEx( m_procHandle, NULL, imagePath.c_str(), NULL, 0, 0, NULL, 0);
+        if (!moduleBase)
+        {
+            std::cerr << "SymLoadModuleEx() failed for module " << imagePath
+                      << ": " << GetLastErrorString() << std::endl;
+            return false;
+        }
+
+        for (size_t i = 0; i < inputEntryList.size(); ++i)
+        {
+            uint64_t offset = inputEntryList[i].symbolOffset;
+            DWORD64 address = moduleBase + offset;
+
+            SYMBOL_INFO* symbolInfo = (SYMBOL_INFO*)s_symbolResolutionBuffer;
+            symbolInfo->SizeOfStruct = sizeof(SYMBOL_INFO);
+            symbolInfo->MaxNameLen = MAX_SYM_NAME;
+
+            SymbolEntry newEntry;
+
+            if ( SymFromAddr( m_procHandle, address, NULL, symbolInfo ) )
+            {
+                newEntry.name = symbolInfo->Name;
+                //std::cout << "Resolved symbol to: '" << newEntry.name << "'" << std::endl;
+            }
+            else
+            {
+                newEntry.name = "[unknown] + " + std::to_string(offset);
+            }
+
+            IMAGEHLP_LINE lineInfo = { 0 };
+            lineInfo.SizeOfStruct = sizeof(IMAGEHLP_LINE64);
+            DWORD displaceMent = 0;
+            if ( SymGetLineFromAddr64( m_procHandle, address, &displaceMent, &lineInfo ) )
+            {
+                newEntry.file = lineInfo.FileName;
+                newEntry.line = int(lineInfo.LineNumber);
+                ///std::cout << "\tline_file: " lineInfo.FileName << ":" << int(lineInfo.LineNumber) << std::endl;
+            }
+
+            resolvedEntries.push_back(std::move(newEntry));
+        }
+
+        SymUnloadModule64(m_procHandle, moduleBase);
+        return true;
+    }
+
+private:
+    static const size_t symbolResolutionBufferSize = sizeof(SYMBOL_INFO) + MAX_SYM_NAME;
+    static char s_symbolResolutionBuffer[symbolResolutionBufferSize];
+
+    std::string GetLastErrorString()
+    {
+        DWORD error = GetLastError();
+        if (error == 0)
+        {
+            return "";
+        }
+
+        LPSTR messageBuffer = nullptr;
+        DWORD dwFlags = FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS;
+        size_t size = FormatMessageA( dwFlags, NULL, error, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                                     (LPSTR)&messageBuffer, 0, NULL );
+
+        std::string message(messageBuffer, size);
+        LocalFree(messageBuffer);
+
+        return message;
+    }
+
+    HANDLE m_procHandle = nullptr;
+};
+
+char SymbolResolver::s_symbolResolutionBuffer[symbolResolutionBufferSize];
+
+
+SymbolResolver* CreateResolver()
+{
+    SymbolResolver* resolver = new SymbolResolver();
+    return resolver;
+}
+
+void DestroySymbolResolver(SymbolResolver* resolver)
+{
+    delete resolver;
+}
```

Comment:
Why is `new` and `delete` relegated to functions here?

Reply:
The code was a bit different before when I had more state in SymbolResolver which is a private type in the CPP. There is no need for this right, removed it.